test-framework/junit5/src/main/java/io/quarkus/test/junit/internal/VerifyMockitoMocksCallback.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.test.junit.internal;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.quarkus.test.InjectMock;
+import io.quarkus.test.junit.callback.QuarkusTestAfterConstructCallback;
+
+public class VerifyMockitoMocksCallback implements QuarkusTestAfterConstructCallback {
+
+    @Override
+    public void afterConstruct(Object testInstance) {
+        Class<? extends Annotation> deprecatedInjectMock = loadDeprecatedInjectMock();
+        if (deprecatedInjectMock == null) {
+            List<Field> injectMockFields = new ArrayList<>();
+            Class<?> current = testInstance.getClass();
+            while (current.getSuperclass() != null) {
+                for (Field field : current.getDeclaredFields()) {
+                    if (field.isAnnotationPresent(InjectMock.class)) {
+                        injectMockFields.add(field);
+                    }
+                }
+            }
+            current = current.getSuperclass();
+            if (!injectMockFields.isEmpty()) {
+                throw new IllegalStateException(
```

Comment:
We need this because the `@io.quarkus.test.InjectMock` annotation is on the class path even if not supported (i.e. no `quarkus-junit5-mockito` dependency)...